packages/react/src/BlockNoteTheme.ts

```diff
@@ -1,82 +1,110 @@
 import { MantineThemeOverride } from "@mantine/core";
+import { defaultTheme } from "./defaultTheme";
 
-type ColorScheme = [
-  string,
-  string,
-  string,
-  string,
-  string,
-  string,
-  string,
-  string,
-  string,
-  string
-];
+export type Color = {
+  light: string;
+  dark: string;
+};
 
-export const blockNoteColorScheme: ColorScheme = [
-  "#FFFFFF",
-  "#EFEFEF",
-  "#CFCFCF",
-  "#AFAFAF",
-  "#7F7F7F",
-  "#3F3F3F",
-  "#1F1F1F",
-  "#161616",
-  "#0F0F0F",
-  "#000000",
-];
+export type CombinedColor = {
+  text: Color;
+  background: Color;
+};
 
-export const getBlockNoteTheme = (
-  useDarkTheme: boolean = false
-): MantineThemeOverride => {
-  const boxShadow = `0px 4px 8px ${
-    useDarkTheme ? blockNoteColorScheme[8] : blockNoteColorScheme[2]
-  }, 0px 0px 1px ${
-    useDarkTheme ? blockNoteColorScheme[6] : blockNoteColorScheme[1]
-  }`;
+export type ColorScheme = {
+  editor: CombinedColor;
+  menu: CombinedColor;
+  tooltip: CombinedColor;
+  hovered: CombinedColor;
+  selected: CombinedColor;
+  disabled: CombinedColor;
+  shadow: Color;
+  border: Color;
+  sideMenu: Color;
+  highlightColors: {
+    gray: CombinedColor;
+    brown: CombinedColor;
+    red: CombinedColor;
+    orange: CombinedColor;
+    yellow: CombinedColor;
+    green: CombinedColor;
+    blue: CombinedColor;
+    purple: CombinedColor;
+    pink: CombinedColor;
+  };
+};
 
-  const border = `1px solid ${
-    useDarkTheme ? blockNoteColorScheme[7] : blockNoteColorScheme[1]
-  }`;
+export type Theme = {
+  colors: ColorScheme;
+  borderRadius: number;
+  fontFamily: string;
+};
 
-  const sideMenu = useDarkTheme
-    ? blockNoteColorScheme[4]
-    : blockNoteColorScheme[2];
+export const blockNoteToMantineTheme = (
```

Comment:
My suggestion for dark / light would also make the code in this method a lot easier